src/site/_includes/components/component-preview/macro.njk

```diff
@@ -3,7 +3,7 @@
 {%- macro guidanceComponentPreview(params) -%}
 
 {%- set exampleTemplatePath = "moduk/components/" + params.componentName + "/__examples__/" + params.exampleName + ".njk" -%}
-{%- set idPrefix = params.componentName + "--" + params.exampleName -%}
+{%- set idPrefix = (params.idPrefix or '') + params.componentName + "--" + params.exampleName -%}
```

Comment:
Looks good – think it might be better as a suffix though and with a separator. Something like this maybe (not sure if it can be simplified):

```suggestion
{%- if params.idSuffix -%}
  {%- set idSuffix = "--" + params.idSuffix -%}
{%- else -%}
  {%- set idSuffix = "" -%}
{%- endif -%}

{%- set idPrefix = (params.idPrefix or '') + params.componentName + "--" + params.exampleName + idSuffix -%}
```